### DISCUSSION TITLE
Why disable `delete-active-region` in meow.

### DISCUSSION THREAD

**Discussion by roife** [meow-edit/meow#298] [General]
By default, the variable `delete-active-region` is set to `true`, so when I select a region and press `DEL`, the whole selected region will be deleted.

However, `meow` set `delete-active-region` to `nil`. It makes `DEL` just delete a single character under `insert` mode even with region active, which is quite strange.

I wonder why `meow` set `delete-active-region` to `false` by default and how to delete the selected region under `insert` mode.

**Comment by eshrh** [comment]
I believe it's because internally, the meow key for deletion doesn't do anything but call the command bound to `C-d`. 

If this were to be set to true, then meow-delete would delete the whole active region, even in normal mode! There would be no distinction between meow-kill and meow-delete. Meow-delete kind of exists for the purpose of one character deletion. Setting the variable to nil and binding to C-d is just a low-overhead approach.

Do you have a specific use case in mind for deleting the region with the same key as delete-char instead of binding delete-region to a key?

**Reply by roife** [reply]
In most editors and OS default input boxes (except for CLI editors such as Vim), when use `DEL` on a active region, the region will be deleted. So it is strange that I cannot do so when using Meow. Maybe the variable should be set to `true` temporarily before entering `insert` mode and set to `nil` after existing `insert` mode, which is more intuitive.

**Comment by DogLooksGood** [comment]
The idea is in meow you have one command for region operation and one command for non-region operation. Like `meow-kill` and `meow-delete`, so you don't have to care whether you have a region or not.
